src/dosext/mfs/lfn.c

```diff
@@ -697,28 +697,21 @@ static int wildcard_delete(char *fpath, int drive)
 			strcpy(slash + 1, de->d_long_name);
 
 			d_printf("LFN: wildcard delete '%s'\n", fpath);
-			stat(fpath, &st);
-			/* don't remove directories */
-			if (st.st_mode & S_IFDIR)
-				continue;
 
+			/* don't remove directories */
 			if ((dirattr >> 8) & DIRECTORY)
 				continue;
-#if 0
-			if (access(fpath, W_OK) == -1) {
-				errcode = EACCES;
-			} else {
-				errcode = unlink(fpath) ? errno : 0;
-			}
-#else
+
 			errcode = unlink(fpath);
-#endif
 			if (errcode != 0) {
+				if (errno == EISDIR)
+					continue;
+
 				d_printf("LFN: Delete failed(%s) '%s'\n",
 					strerror(errno), fpath);
 				free(pattern);
 				dos_closedir(dir);
-				if (errcode == EACCES) {
+				if (errno == EACCES) {
```

Comment:
Are you sure about that?
errno could have changed.

Reply:
Okay, yes I see, thanks. I've also changed it now to make it more obvious which errors are unix and which dos.